## Final Project Submission

Please fill out:
* Student name: Joanne Kariuki
* Student pace: self paced / part time / full time: Part time
* Scheduled project review date/time: Wednesday, 12th February 2025
* Instructor name: Antonny Muiko
* Blog post URL:


In [79]:
# Your code here - remember to use markdown cells for comments as well!

import os

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# Read CSV file to Pandas DataFrame


In [5]:
# Step 1: get the path to the dataframe

Aviation_data = pd.read_csv('data/Aviation_Data.csv', header=0)

Aviation_data.head()

C:\Users\x1 carbon\AppData\Local\Temp\ipykernel_38620\1952344942.py:3: DtypeWarning: Columns (6,7,28) have mixed types. Specify dtype option on import or set low_memory=False.
  Aviation_data = pd.read_csv('data/Aviation_Data.csv', header=0)


Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country   Latitude  Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States        NaN        NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN        NaN          NaN   
2    Saltville, VA  United States  36.922223 -81.878056          NaN   
3       EUREKA, CA  United States        NaN        NaN          NaN   
4       Canton, OH  United States        NaN        NaN          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   
3          NaN  ...          Personal         NaN                  2.0   
4          NaN  ...          Personal         NaN                  1.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   
3                    0.0                  0.0             0.0   
4                    2.0                  NaN             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  
3               IMC                 Cruise  Probable Cause       12-09-2000  
4               VMC               Approach  Probable Cause       16-04-1980  

[5 rows x 31 columns]

### Checking the tail of the data

In [36]:
Aviation_data.tail()

Event.Id Investigation.Type Accident.Number  Event.Date  \
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

            Location        Country Latitude Longitude Airport.Code  \
90343  Annapolis, MD  United States      NaN       NaN          NaN   
90344    Hampton, NH  United States      NaN       NaN          NaN   
90345     Payson, AZ  United States  341525N  1112021W          PAN   
90346     Morgan, UT  United States      NaN       NaN          NaN   
90347     Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
90343          NaN  ...          Personal                 NaN   
90344          NaN  ...               NaN                 NaN   
90345       PAYSON  ...          Personal                 NaN   
90346          NaN  ...          Personal  MC CESSNA 210N LLC   
90347          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
90343                  0.0                    1.0                  0.0   
90344                  0.0                    0.0                  0.0   
90345                  0.0                    0.0                  0.0   
90346                  0.0                    0.0                  0.0   
90347                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight Report.Status  \
90343             0.0               NaN                    NaN           NaN   
90344             0.0               NaN                    NaN           NaN   
90345             1.0               VMC                    NaN           NaN   
90346             0.0               NaN                    NaN           NaN   
90347             1.0               NaN                    NaN           NaN   

      Publication.Date  
90343       29-12-2022  
90344              NaN  
90345       27-12-2022  
90346              NaN  
90347       30-12-2022  

[5 rows x 31 columns]

### Brief description of the data

In [6]:
Aviation_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90348 entries, 0 to 90347
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      90348 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50132 non-null  object 
 9   Airport.Name            52704 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87507 non-null  object 
 14  Make                    88826 non-null

### List of data column names

In [7]:
Aviation_data.columns

Index(['Event.Id', 'Investigation.Type', 'Accident.Number', 'Event.Date',
       'Location', 'Country', 'Latitude', 'Longitude', 'Airport.Code',
       'Airport.Name', 'Injury.Severity', 'Aircraft.damage',
       'Aircraft.Category', 'Registration.Number', 'Make', 'Model',
       'Amateur.Built', 'Number.of.Engines', 'Engine.Type', 'FAR.Description',
       'Schedule', 'Purpose.of.flight', 'Air.carrier', 'Total.Fatal.Injuries',
       'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured',
       'Weather.Condition', 'Broad.phase.of.flight', 'Report.Status',
       'Publication.Date'],
      dtype='object')

### Numeric columns description

In [37]:
Aviation_data.describe()

Number.of.Engines  Total.Fatal.Injuries  Total.Serious.Injuries  \
count       82805.000000          77488.000000            76379.000000   
mean            1.146585              0.647855                0.279881   
std             0.446510              5.485960                1.544084   
min             0.000000              0.000000                0.000000   
25%             1.000000              0.000000                0.000000   
50%             1.000000              0.000000                0.000000   
75%             1.000000              0.000000                0.000000   
max             8.000000            349.000000              161.000000   

       Total.Minor.Injuries  Total.Uninjured  
count          76956.000000     82977.000000  
mean               0.357061         5.325440  
std                2.235625        27.913634  
min                0.000000         0.000000  
25%                0.000000         0.000000  
50%                0.000000         1.000000  
75%                0.000000         2.000000  
max              380.000000       699.000000

# Data Cleaning


### Checking if there are any duplicate rows

In [8]:
Aviation_data.duplicated().any()

np.True_

### Number of duplicated rows

In [9]:
Aviation_data.duplicated().sum()

np.int64(1390)

### Drop the duplicates

In [10]:
Aviation_data.drop_duplicates(keep='first', inplace=True)

### Proof there are no more duplicates

In [11]:
Aviation_data.duplicated().any()

np.False_

## Check for missing values

In [23]:
Aviation_data.isnull().any()

Event.Id                   True
Investigation.Type        False
Accident.Number            True
Event.Date                 True
Location                   True
Country                    True
Latitude                   True
Longitude                  True
Airport.Code               True
Airport.Name               True
Injury.Severity            True
Aircraft.damage            True
Aircraft.Category          True
Registration.Number        True
Make                       True
Model                      True
Amateur.Built              True
Number.of.Engines          True
Engine.Type                True
FAR.Description            True
Schedule                   True
Purpose.of.flight          True
Air.carrier                True
Total.Fatal.Injuries       True
Total.Serious.Injuries     True
Total.Minor.Injuries       True
Total.Uninjured            True
Weather.Condition          True
Broad.phase.of.flight      True
Report.Status              True
Publication.Date           True
dtype: b

### Count of missing values

In [29]:
missing = Aviation_data.isnull().sum()

missing.sort_values(ascending=False)

Schedule                  76376
Air.carrier               72310
FAR.Description           56935
Aircraft.Category         56671
Longitude                 54585
Latitude                  54576
Airport.Code              38826
Airport.Name              36254
Broad.phase.of.flight     27234
Publication.Date          15299
Total.Serious.Injuries    12579
Total.Minor.Injuries      12002
Total.Fatal.Injuries      11470
Engine.Type                7165
Report.Status              6453
Purpose.of.flight          6261
Number.of.Engines          6153
Total.Uninjured            5981
Weather.Condition          4561
Aircraft.damage            3263
Registration.Number        1451
Injury.Severity            1069
Country                     295
Amateur.Built               171
Model                       161
Make                        132
Location                    121
Event.Date                   69
Event.Id                     69
Accident.Number              69
Investigation.Type            0
dtype: i

### The percentage of the missing data

In [30]:
percentage_missing = np.round(((Aviation_data.isnull().sum() / len(Aviation_data)) * 100), 2)

percentage_missing. sort_values(ascending=False)

Schedule                  85.86
Air.carrier               81.29
FAR.Description           64.00
Aircraft.Category         63.71
Longitude                 61.36
Latitude                  61.35
Airport.Code              43.65
Airport.Name              40.75
Broad.phase.of.flight     30.61
Publication.Date          17.20
Total.Serious.Injuries    14.14
Total.Minor.Injuries      13.49
Total.Fatal.Injuries      12.89
Engine.Type                8.05
Report.Status              7.25
Purpose.of.flight          7.04
Number.of.Engines          6.92
Total.Uninjured            6.72
Weather.Condition          5.13
Aircraft.damage            3.67
Registration.Number        1.63
Injury.Severity            1.20
Country                    0.33
Amateur.Built              0.19
Model                      0.18
Make                       0.15
Location                   0.14
Event.Date                 0.08
Event.Id                   0.08
Accident.Number            0.08
Investigation.Type         0.00
dtype: f

### Store the series data in a dataframe

In [ ]:
miss_aviation_data = pd.DataFrame({
    "Count": missing.values,
    "%": percentage_missing.values
} , index = missing.index)

# drop values where count is 0
miss_aviation_data.drop(miss_aviation_data[miss_aviation_data['Count'] == 0].index, inplace=True)

# sort values in descending order
miss_aviation_data.sort_values(by='Count', ascending=False, inplace=True)


miss_aviation_data

Count      %
Schedule                76376  85.86
Air.carrier             72310  81.29
FAR.Description         56935  64.00
Aircraft.Category       56671  63.71
Longitude               54585  61.36
Latitude                54576  61.35
Airport.Code            38826  43.65
Airport.Name            36254  40.75
Broad.phase.of.flight   27234  30.61
Publication.Date        15299  17.20
Total.Serious.Injuries  12579  14.14
Total.Minor.Injuries    12002  13.49
Total.Fatal.Injuries    11470  12.89
Engine.Type              7165   8.05
Report.Status            6453   7.25
Purpose.of.flight        6261   7.04
Number.of.Engines        6153   6.92
Total.Uninjured          5981   6.72
Weather.Condition        4561   5.13
Aircraft.damage          3263   3.67
Registration.Number      1451   1.63
Injury.Severity          1069   1.20
Country                   295   0.33
Amateur.Built             171   0.19
Model                     161   0.18
Make                      132   0.15
Location                  121   0.14
Accident.Number            69   0.08
Event.Date                 69   0.08
Event.Id                   69   0.08

### Drop the columns with more than 31% of missing data

In [83]:
columns_to_drop = ['Schedule', 'Air.carrier', 'FAR.Description', 'Aircraft.Category', 'Longitude', 'Latitude', 'Airport.Code', 'Airport.Name']	

Aviation_data_clean = Aviation_data.drop(columns=columns_to_drop)

Aviation_data_clean.head()

Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country Injury.Severity Aircraft.damage  \
0  MOOSE CREEK, ID  United States        Fatal(2)       Destroyed   
1   BRIDGEPORT, CA  United States        Fatal(4)       Destroyed   
2    Saltville, VA  United States        Fatal(3)       Destroyed   
3       EUREKA, CA  United States        Fatal(2)       Destroyed   
4       Canton, OH  United States        Fatal(1)       Destroyed   

  Registration.Number      Make  ...    Engine.Type Purpose.of.flight  \
0              NC6404   Stinson  ...  Reciprocating          Personal   
1              N5069P     Piper  ...  Reciprocating          Personal   
2              N5142R    Cessna  ...  Reciprocating          Personal   
3              N1168J  Rockwell  ...  Reciprocating          Personal   
4               N15NY    Cessna  ...            NaN          Personal   

   Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                   2.0                    0.0                  0.0   
1                   4.0                    0.0                  0.0   
2                   3.0                    NaN                  NaN   
3                   2.0                    0.0                  0.0   
4                   1.0                    2.0                  NaN   

   Total.Uninjured  Weather.Condition  Broad.phase.of.flight   Report.Status  \
0              0.0                UNK                 Cruise  Probable Cause   
1              0.0                UNK                Unknown  Probable Cause   
2              NaN                IMC                 Cruise  Probable Cause   
3              0.0                IMC                 Cruise  Probable Cause   
4              0.0                VMC               Approach  Probable Cause   

  Publication.Date  
0              NaN  
1       19-09-1996  
2       26-02-2007  
3       12-09-2000  
4       16-04-1980  

[5 rows x 23 columns]

### Identify the most important columns

* Make
* Model
* Event.Date
* Purpose.of.flight
* Weather.Conditions
* Broad.face.of.flight
* Total.Fatal.Injuries
* Total.Minor.Injuries
* Total.Serious.Injuries
* Total.Uninjured

### Replace the missing values in column 'Total.Fatal.Injuries'with mean

In [97]:
mean_fatal_injuries = Aviation_data_clean['Total.Fatal.Injuries'].mean()

Aviation_data_clean['Total.Fatal.Injuries'] = Aviation_data_clean['Total.Fatal.Injuries'].fillna(mean_fatal_injuries)


### Replace the missing values in column 'Total.Minor.Injuries'with mean

In [98]:
mean_minor_injuries = Aviation_data_clean['Total.Minor.Injuries'].mean()

Aviation_data_clean['Total.Minor.Injuries'] = Aviation_data_clean['Total.Minor.Injuries'].fillna(mean_minor_injuries)


### Replace the missing values in column 'Total.Serious.Injuries' with mean

In [99]:
mean_serious_injuries = Aviation_data_clean['Total.Serious.Injuries'].mean()

Aviation_data_clean['Total.Serious.Injuries'] = Aviation_data_clean['Total.Serious.Injuries'].fillna(mean_serious_injuries)



### Replace the missing values in column 'Total.Uninjured' with mean

In [101]:
mean_uninjured = Aviation_data_clean['Total.Uninjured'].mean()

Aviation_data_clean['Total.Uninjured'] = Aviation_data_clean['Total.Uninjured'].fillna(mean_uninjured)


### Replace the missing values in column 'Make' with mode

In [116]:
mode_make= Aviation_data_clean['Make'].mode()[0]

Aviation_data_clean['Make'] = Aviation_data_clean['Make'].fillna(mode_make)


### Replace the missing values in column 'Model' with mode

In [109]:
mode_model= Aviation_data_clean['Model'].mode()[0]

Aviation_data_clean['Model'] = Aviation_data_clean['Model'].fillna(mode_model)

### Replace the missing values in column 'Purpose.of.flight' with mode

In [112]:
mode_purpose_of_flight = Aviation_data_clean['Purpose.of.flight'].mode()[0]

Aviation_data_clean['Purpose.of.flight'] = Aviation_data_clean['Purpose.of.flight'].fillna(mode_purpose_of_flight)

### Replace the missing values in column 'Broad.phase.of.flight' with mode

In [113]:
mode_phase_of_flight = Aviation_data_clean['Broad.phase.of.flight'].mode()[0]

Aviation_data_clean['Broad.phase.of.flight'] = Aviation_data_clean['Broad.phase.of.flight'].fillna(mode_phase_of_flight)

### Replace the missing values in column 'Weather.Condition' with mode

In [114]:
mode_weather_condition = Aviation_data_clean['Weather.Condition'].mode()[0]

Aviation_data_clean['Weather.Condition'] = Aviation_data_clean['Weather.Condition'].fillna(mode_weather_condition)

### Unify the date format of 'Event.Date'

In [119]:
Aviation_data_clean['Event.Date'] = pd.to_datetime(Aviation_data_clean['Event.Date'], format='%Y-%m-%d')

Aviation_data_clean.head()

Event.Id Investigation.Type Accident.Number Event.Date  \
0  20001218X45444           Accident      SEA87LA080 1948-10-24   
1  20001218X45447           Accident      LAX94LA336 1962-07-19   
2  20061025X01555           Accident      NYC07LA005 1974-08-30   
3  20001218X45448           Accident      LAX96LA321 1977-06-19   
4  20041105X01764           Accident      CHI79FA064 1979-08-02   

          Location        Country Injury.Severity Aircraft.damage  \
0  MOOSE CREEK, ID  United States        Fatal(2)       Destroyed   
1   BRIDGEPORT, CA  United States        Fatal(4)       Destroyed   
2    Saltville, VA  United States        Fatal(3)       Destroyed   
3       EUREKA, CA  United States        Fatal(2)       Destroyed   
4       Canton, OH  United States        Fatal(1)       Destroyed   

  Registration.Number      Make  ...    Engine.Type Purpose.of.flight  \
0              NC6404   Stinson  ...  Reciprocating          Personal   
1              N5069P     Piper  ...  Reciprocating          Personal   
2              N5142R    Cessna  ...  Reciprocating          Personal   
3              N1168J  Rockwell  ...  Reciprocating          Personal   
4               N15NY    Cessna  ...            NaN          Personal   

   Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
0                   2.0               0.000000             0.000000   
1                   4.0               0.000000             0.000000   
2                   3.0               0.279881             0.357061   
3                   2.0               0.000000             0.000000   
4                   1.0               2.000000             0.357061   

   Total.Uninjured  Weather.Condition  Broad.phase.of.flight   Report.Status  \
0          0.00000                UNK                 Cruise  Probable Cause   
1          0.00000                UNK                Unknown  Probable Cause   
2          5.32544                IMC                 Cruise  Probable Cause   
3          0.00000                IMC                 Cruise  Probable Cause   
4          0.00000                VMC               Approach  Probable Cause   

  Publication.Date  
0              NaN  
1       19-09-1996  
2       26-02-2007  
3       12-09-2000  
4       16-04-1980  

[5 rows x 23 columns]

### Replace missing dates

In [123]:
Aviation_data_clean['Event.Date'].ffill()

fill_dates = Aviation_data_clean['Event.Date'].ffill()

Aviation_data_clean['Event.Date'] = Aviation_data_clean['Event.Date'].fillna(fill_dates)

### Separate the years from the date/month and create its own column

In [125]:
Aviation_data_clean.loc[:, ('Event.Year')] = Aviation_data_clean.loc[:, ('Event.Date')].dt.year
Aviation_data_clean['Event.Year'].unique()

array([1948., 1962., 1974., 1977., 1979., 1981., 1982., 1983., 1984.,
       1985., 1986., 1987., 1988., 1989., 1990., 1991., 1992., 1993.,
       1994., 1995., 1996., 1997., 1998., 1999., 2000., 2001., 2002.,
       2003., 2004., 2005., 2006., 2007., 2008., 2009., 2010., 2011.,
       2012., 2013., 2014., 2015., 2016., 2017., 2018., 2019., 2020.,
       2021., 2022.])

### confirming the columns to be used for visualization have no missing values

In [124]:
Aviation_data_clean.isnull().any()

Event.Id                   True
Investigation.Type        False
Accident.Number            True
Event.Date                False
Location                   True
Country                    True
Injury.Severity            True
Aircraft.damage            True
Registration.Number        True
Make                      False
Model                     False
Amateur.Built              True
Number.of.Engines          True
Engine.Type                True
Purpose.of.flight         False
Total.Fatal.Injuries      False
Total.Serious.Injuries    False
Total.Minor.Injuries      False
Total.Uninjured           False
Weather.Condition         False
Broad.phase.of.flight     False
Report.Status              True
Publication.Date           True
Event.Year                 True
dtype: bool

# Outliers

As per my numerical data description the max value for passangers(injurd or uninjured) is 699, while the highest capacity in an aiplane is 853 passangers; hence an outlier analysis won't be necessary.

## Save to CSV

In [143]:
Aviation_data_clean.to_csv('Clean_aviation_data.csv', index=False, na_rep='NULL')

## Check for duplicates in unique entry 'Event.Id'

In [133]:
duplicates = Aviation_data_clean[Aviation_data_clean.duplicated(subset=['Event.Id'], keep=False)]

duplicates

Event.Id Investigation.Type Accident.Number Event.Date  \
117    20020917X01908           Accident     DCA82AA012B 1982-01-19   
118    20020917X01908           Accident     DCA82AA012A 1982-01-19   
153    20020917X02259           Accident     LAX82FA049A 1982-01-23   
158    20020917X02400           Accident     MIA82FA038B 1982-01-23   
159    20020917X02400           Accident     MIA82FA038A 1982-01-23   
...               ...                ...             ...        ...   
90255  20221121106336           Accident      WPR23LA041 2022-11-18   
90256  20221122106340           Incident      DCA23WA071 2022-11-18   
90257  20221122106340           Incident      DCA23WA071 2022-11-18   
90272  20221123106354           Accident      WPR23LA045 2022-11-22   
90273  20221123106354           Accident      WPR23LA045 2022-11-22   

                 Location        Country Injury.Severity Aircraft.damage  \
117          ROCKPORT, TX  United States        Fatal(3)       Destroyed   
118          ROCKPORT, TX  United States        Fatal(3)       Destroyed   
153       VICTORVILLE, CA  United States        Fatal(2)       Destroyed   
158    NEWPORT RICHEY, FL  United States       Non-Fatal     Substantial   
159    NEWPORT RICHEY, FL  United States       Non-Fatal     Substantial   
...                   ...            ...             ...             ...   
90255       Las Vegas, NV  United States       Non-Fatal     Substantial   
90256         Marrakech,         Morocco             NaN             NaN   
90257         Marrakech,         Morocco             NaN             NaN   
90272       San Diego, CA  United States       Non-Fatal     Substantial   
90273       San Diego, CA  United States       Non-Fatal     Substantial   

      Registration.Number        Make  ...          Purpose.of.flight  \
117                N26660     Grumman  ...                   Personal   
118                N336SA  Swearingen  ...        Executive/corporate   
153                N7860V      Mooney  ...                   Personal   
158                N45453      Cessna  ...                   Personal   
159                N32555       Piper  ...                   Personal   
...                   ...         ...  ...                        ...   
90255                N88E      CESSNA  ...              Instructional   
90256              G-MIDT      AIRBUS  ...                   Personal   
90257              CS-DVH      CESSNA  ...                   Personal   
90272              N160AQ    SIKORSKY  ...              Instructional   
90273              166583    SIKORSKY  ...  Public Aircraft - Federal   

      Total.Fatal.Injuries  Total.Serious.Injuries Total.Minor.Injuries  \
117                    3.0                     0.0                  0.0   
118                    3.0                     0.0                  0.0   
153                    2.0                     0.0                  4.0   
158                    0.0                     0.0                  0.0   
159                    0.0                     0.0                  0.0   
...                    ...                     ...                  ...   
90255                  0.0                     0.0                  0.0   
90256                  0.0                     0.0                  0.0   
90257                  0.0                     0.0                  0.0   
90272                  0.0                     0.0                  0.0   
90273                  0.0                     0.0                  0.0   

      Total.Uninjured  Weather.Condition  Broad.phase.of.flight  \
117               0.0                IMC               Approach   
118               0.0                IMC               Approach   
153               0.0                VMC                Unknown   
158               3.0                VMC                 Cruise   
159               3.0                VMC               Approach   
...               ...                ...                    ...   


### Count of the duplicated data

In [134]:

Aviation_data_clean.duplicated(subset='Event.Id').sum()

np.int64(1006)

### % of duplicates in the column

In [135]:
# % of duplicates in the column
np.round(Aviation_data_clean.duplicated(subset='Event.Id').sum() / len(Aviation_data_clean), 3)

np.float64(0.011)

### Get the index of rows with duplicate Event.Id

In [ ]:

duplicated_indices = Aviation_data_clean[Aviation_data_clean['Event.Id'].duplicated(keep=False)].index

duplicated_indices

Index([  117,   118,   153,   158,   159,   160,   242,   244,   245,   248,
       ...
       90050, 90090, 90235, 90236, 90254, 90255, 90256, 90257, 90272, 90273],
      dtype='int64', length=1943)

### filter the data

In [ ]:

filtered_Aviation_data = Aviation_data_clean.loc[duplicated_indices]
filtered_Aviation_data.shape

filtered_Aviation_data

Event.Id Investigation.Type Accident.Number Event.Date  \
117    20020917X01908           Accident     DCA82AA012B 1982-01-19   
118    20020917X01908           Accident     DCA82AA012A 1982-01-19   
153    20020917X02259           Accident     LAX82FA049A 1982-01-23   
158    20020917X02400           Accident     MIA82FA038B 1982-01-23   
159    20020917X02400           Accident     MIA82FA038A 1982-01-23   
...               ...                ...             ...        ...   
90255  20221121106336           Accident      WPR23LA041 2022-11-18   
90256  20221122106340           Incident      DCA23WA071 2022-11-18   
90257  20221122106340           Incident      DCA23WA071 2022-11-18   
90272  20221123106354           Accident      WPR23LA045 2022-11-22   
90273  20221123106354           Accident      WPR23LA045 2022-11-22   

                 Location        Country Injury.Severity Aircraft.damage  \
117          ROCKPORT, TX  United States        Fatal(3)       Destroyed   
118          ROCKPORT, TX  United States        Fatal(3)       Destroyed   
153       VICTORVILLE, CA  United States        Fatal(2)       Destroyed   
158    NEWPORT RICHEY, FL  United States       Non-Fatal     Substantial   
159    NEWPORT RICHEY, FL  United States       Non-Fatal     Substantial   
...                   ...            ...             ...             ...   
90255       Las Vegas, NV  United States       Non-Fatal     Substantial   
90256         Marrakech,         Morocco             NaN             NaN   
90257         Marrakech,         Morocco             NaN             NaN   
90272       San Diego, CA  United States       Non-Fatal     Substantial   
90273       San Diego, CA  United States       Non-Fatal     Substantial   

      Registration.Number        Make  ...          Purpose.of.flight  \
117                N26660     Grumman  ...                   Personal   
118                N336SA  Swearingen  ...        Executive/corporate   
153                N7860V      Mooney  ...                   Personal   
158                N45453      Cessna  ...                   Personal   
159                N32555       Piper  ...                   Personal   
...                   ...         ...  ...                        ...   
90255                N88E      CESSNA  ...              Instructional   
90256              G-MIDT      AIRBUS  ...                   Personal   
90257              CS-DVH      CESSNA  ...                   Personal   
90272              N160AQ    SIKORSKY  ...              Instructional   
90273              166583    SIKORSKY  ...  Public Aircraft - Federal   

      Total.Fatal.Injuries  Total.Serious.Injuries Total.Minor.Injuries  \
117                    3.0                     0.0                  0.0   
118                    3.0                     0.0                  0.0   
153                    2.0                     0.0                  4.0   
158                    0.0                     0.0                  0.0   
159                    0.0                     0.0                  0.0   
...                    ...                     ...                  ...   
90255                  0.0                     0.0                  0.0   
90256                  0.0                     0.0                  0.0   
90257                  0.0                     0.0                  0.0   
90272                  0.0                     0.0                  0.0   
90273                  0.0                     0.0                  0.0   

      Total.Uninjured  Weather.Condition  Broad.phase.of.flight  \
117               0.0                IMC               Approach   
118               0.0                IMC               Approach   
153               0.0                VMC                Unknown   
158               3.0                VMC                 Cruise   
159               3.0                VMC               Approach   
...               ...                ...                    ...   


### Flag duplicates for Manual Review

In [ ]:

filtered_Aviation_data['Is_Duplicate'] = filtered_Aviation_data.duplicated('Event.Id', keep=False)
print(filtered_Aviation_data[filtered_Aviation_data['Is_Duplicate']])

             Event.Id Investigation.Type Accident.Number Event.Date  \
117    20020917X01908           Accident     DCA82AA012B 1982-01-19   
118    20020917X01908           Accident     DCA82AA012A 1982-01-19   
153    20020917X02259           Accident     LAX82FA049A 1982-01-23   
158    20020917X02400           Accident     MIA82FA038B 1982-01-23   
159    20020917X02400           Accident     MIA82FA038A 1982-01-23   
...               ...                ...             ...        ...   
90255  20221121106336           Accident      WPR23LA041 2022-11-18   
90256  20221122106340           Incident      DCA23WA071 2022-11-18   
90257  20221122106340           Incident      DCA23WA071 2022-11-18   
90272  20221123106354           Accident      WPR23LA045 2022-11-22   
90273  20221123106354           Accident      WPR23LA045 2022-11-22   

                 Location        Country Injury.Severity Aircraft.damage  \
117          ROCKPORT, TX  United States        Fatal(3)       Destroye

# Data Visualization


### Plot data for Make vs Aircraft damage

In [ ]:
fig, ax = plt.Subplot(figsize=(8,4))
ax.plot